src/components/v5/common/ActionSidebar/partials/UserSelect/hooks.ts

```diff
@@ -1,28 +1,52 @@
+import { useMemo } from 'react';
 import { useColonyContext } from '~hooks';
 import { useGetColonyMembers } from '~v5/shared/MembersSelect/hooks';
 import { UserSelectHookProps } from './types';
+import { useGetVerifiedMembersQuery } from '~gql';
+import { SearchSelectOption } from '~v5/shared/SearchSelect/types';
 
-export const useUserSelect = (): UserSelectHookProps => {
+export const useUserSelect = (inputValue: string): UserSelectHookProps => {
   const { colony } = useColonyContext();
-  const { members, loading } = useGetColonyMembers(colony?.colonyAddress);
+  const { colonyAddress = '', metadata } = colony ?? {};
+  const { members, loading } = useGetColonyMembers(colonyAddress);
+  const { data, loading: verifiedMembersLoading } = useGetVerifiedMembersQuery({
+    variables: { colonyAddress },
+    skip: !colonyAddress,
+  });
+  const isWhitelistActivated = metadata?.isWhitelistActivated;
 
-  const users = members?.map((member) => {
-    const { label, walletAddress, avatar } = member || {};
+  const verifiedUsers: SearchSelectOption[] = useMemo(
+    () =>
+      (data?.getContributorsByColony?.items ?? []).map((item, index) => ({
+        value: item?.user?.walletAddress || '',
+        label: item?.user?.profile?.displayName || '',
+        avatar:
+          item?.user?.profile?.avatar || item?.user?.profile?.thumbnail || '',
+        walletAddress: item?.user?.walletAddress,
+        id: index,
+        showAvatar: true,
+      })),
+    [data?.getContributorsByColony?.items],
+  );
 
-    return {
-      label: label || '',
-      value: walletAddress || '',
-      avatar: avatar || '',
-      walletAddress: walletAddress || '',
-      showAvatar: true,
-    };
-  });
+  const preparedUserOptions: SearchSelectOption[] = isWhitelistActivated
+    ? verifiedUsers
+    : members || [];
+
+  const isUserVerified = preparedUserOptions.some(
+    ({ label, walletAddress }) =>
+      label === inputValue || walletAddress === inputValue,
```

Comment:
I don't think it is correct to compare the label to the field value

Reply:
fixed